### PR TITLE
Internalize mount.Interface.IsMountPointMatch

### PR DESCRIPTION
**Pull request by codenrhoden**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
IsMountPointMatch() had no callers outside of the mount package, and has
internal implementation details. This patch makes it no longer be
public.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
As part of https://github.com/kubernetes/utils/pull/100, it was asked why this method was needed rather than just being a simple `mp.Path == dir`. After looking closer, this method does not have an external callers and does not need to be public.

/assign @msau42

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/util/mount/types/types.go
Comment (msau42) [inline pkg/util/mount/types/types.go]: Do we want to specifically call out implementation specifics in the interface comment? Maybe something more generic like "In some environments, the directory could be renamed if the mount point is stale".

Or, from a completely different angle, does this have to be exposed as an interface method? Or can we completely hide it in the implementation?

Comment (codenrhoden) [reply pkg/util/mount/types/types.go]: yeah, there is where I need help. :) One of the review comments was "why is this needed?" and another was "don't mention implementation details." But the reason it's needed is because the implementation is different on Linux. So we could just call out what's different in the Linux implementations (`mount_linux.go`, `nsenter`), but that leaves the question of what to say here...

But you are right... from a different angle, this doesn't appear needed in the interface. It's only called in one place: https://github.com/kubernetes/kubernetes/blob/d951d19ec4d8ec829e83d146200ea91b664b6fa6/pkg/util/mount/mount.go#L252

It's completely internal. We need different implementations for Windows/Linux, but that doesn't mean it has to be in the interface. I'll explore that.

**Comment by codenrhoden** [comment]
@msau42  See what you think of this approach.

I also changed my mind on basing this PR on top of another PR. I'll just deal with the merge conflicts as they happen. I'd rather make obvious progress as we can...

File: pkg/util/mount/mount_helper_common.go
Comment (msau42) [inline pkg/util/mount/mount_helper_common.go]: This is a bit odd. My other thought was making IsNotMountPoint part of the interface.

Comment (codenrhoden) [reply pkg/util/mount/mount_helper_common.go]: Sure, let me know how you want me to proceed...

Before I went this direction, I did a quick search of the codebase, and saw a lot of places in `pkg/kubelet` and `pkg/volume` that were doing switches on `runtime.GOOS`, so I kind of ran with that precedent. :) It made for more compact code that could just be one common internal function.

Comment (msau42) [reply pkg/util/mount/mount_helper_common.go]: Let's get @saad-ali 's thoughts on this.  I think it's a little odd because we already have os-specific implementations of the mounter.

Comment (hogepodge) [reply pkg/util/mount/mount_helper_common.go]: Are there instances where this code wouldn't be run on Linux, and would that be a bug? I would prefer to err on the side of safety if the mounter code somehow winds up as the default code path.

Comment (saad-ali) [reply pkg/util/mount/mount_helper_common.go]: I agree with the higher level goal here which is remove `IsMountPointMatch(...)` from the Mount interface since it is an implementation detail for Mount.

There are already `mount_helper_windows.go` and `mount_helper_unix.go` files, why not make `isMountPointMatch(...)` arch specific, and put a definition in each of those files?

Comment (codenrhoden) [reply pkg/util/mount/mount_helper_common.go]: I had tried that approach originally, but was running into issues because I thought I needed an  `unsupported` implementation along with Linux and Windows. I had failed to notice that the `mount_helper_unix.go` file is actually `+build !windows`, meaning that it builds on Darwin as well. Therefore anything in there is actually for *both* Linux and Darwin. Before I was getting a redeclaration error.

I've fixed this. Should be good to review again.

**Comment by codenrhoden** [comment]
/hold
I'm holding this because if it got final approval, it would potentially go in before https://github.com/kubernetes/kubernetes/pull/81794, and I *really* want to avoid rebasing that one if I can. Rebasing this one is a snap.

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

File: pkg/util/mount/mount_helper_common.go
Comment (saad-ali) [reply pkg/util/mount/mount_helper_common.go]: > I had failed to notice that the mount_helper_unix.go file is actually +build !windows, meaning that it builds on Darwin as well. Therefore anything in there is actually

Unrelated to this PR, but why are we doing `+build !windows` rather then `+build linux`?

**Comment by codenrhoden** [comment]
/hold

I thought the needs-rebase label would be on this automatically and it wouldn’t get into the merge pool. I was wrong. Can’t rebase until tomorrow morning, will remove hold after that. Apologies for any spam.

**Pull request merged**
